bundles/org.openhab.binding.gce/src/main/java/org/openhab/binding/gce/internal/handler/Ipx800DeviceConnector.java

```diff
@@ -35,156 +40,161 @@
  */
 @NonNullByDefault
 public class Ipx800DeviceConnector extends Thread {
-    private static final int DEFAULT_SOCKET_TIMEOUT_MS = 5000;
-    private static final int DEFAULT_RECONNECT_TIMEOUT_MS = 5000;
+    private static final int DEFAULT_SOCKET_TIMEOUT_MS = 10000;
     private static final int MAX_KEEPALIVE_FAILURE = 3;
-    private static final String ENDL = "\r\n";
 
     private final Logger logger = LoggerFactory.getLogger(Ipx800DeviceConnector.class);
+    private final Random randomizer = new Random();
 
-    private final String hostname;
-    private final int portNumber;
-
-    private Optional<M2MMessageParser> messageParser = Optional.empty();
-    private Optional<Socket> socket = Optional.empty();
-    private Optional<BufferedReader> input = Optional.empty();
-    private Optional<PrintWriter> output = Optional.empty();
+    private final M2MMessageParser parser;
+    private final StatusFileAccessor statusAccessor;
+    private final Ipx800EventListener listener;
+    private final Socket socket;
+    private final BufferedReader input;
+    private final PrintWriter output;
 
     private int failedKeepalive = 0;
     private boolean waitingKeepaliveResponse = false;
+    private boolean interrupted = false;
 
-    public Ipx800DeviceConnector(String hostname, int portNumber, ThingUID uid) {
+    public Ipx800DeviceConnector(String hostname, int portNumber, ThingUID uid, Ipx800EventListener listener)
+            throws UnknownHostException, IOException {
         super("OH-binding-" + uid);
-        this.hostname = hostname;
-        this.portNumber = portNumber;
+        this.listener = listener;
+
+        logger.debug("Connecting to {}:{}...", hostname, portNumber);
+        Socket socket = new Socket(hostname, portNumber);
+        socket.setSoTimeout(DEFAULT_SOCKET_TIMEOUT_MS);
+        this.socket = socket;
+
+        output = new PrintWriter(socket.getOutputStream(), true);
+        input = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+        parser = new M2MMessageParser(listener);
+        statusAccessor = new StatusFileAccessor(hostname);
         setDaemon(true);
     }
 
+    /**
+     *
+     * Stop the
+     * device thread
+     */
+
+    public void dispose() {
+        interrupted = true;
```

Comment:
I wonder if this is the right way of doing this.

Reply:
Heavily tested. It works.